pkg/cmd/roachtest/tests/go_helpers.go

```diff
@@ -43,7 +43,7 @@ func installGolang(
 	}
 
 	if err := repeatRunE(
-		ctx, t, c, node, "download go", `curl -fsSL https://dl.google.com/go/go1.17.6.linux-amd64.tar.gz > /tmp/go.tgz`,
+		ctx, t, c, node, "download go", `curl -fsSL https://dl.google.com/go/go1.17.11.linux-amd64.tar.gz > /tmp/go.tgz`,
 	); err != nil {
 		t.Fatal(err)
 	}
```

Comment:
[This comment](https://github.com/cockroachdb/cockroach/pull/82816/files#r895851508) still applies, you might as well fix it here.